### ISSUE TITLE
`constructor signature check` dosen't work

### ISSUE THREAD

**Issue by ELSS-ZION** [microsoft/TypeScript#38703]
**TypeScript Version:**  Version 4.0.0-dev.20200521

**Search Terms:** constructor signature check

**Code**

```ts
interface ClockConstructor {
    new(h: number, minute: number)
}

function createClock(ctor: ClockConstructor, hour: number, minute: number) {
    return new ctor(hour, minute);
}

class Clock {
    constructor() {
        console.log("Clock")
    }
}

let obj = createClock(Clock, 7, 32);    // Clock's constructor has no params, this statement should trigger a error, but it didn't.
let ccc: ClockConstructor = Clock;       // Clock's constructor has no params, this statement should trigger a error, but it didn't.
```

**Expected behavior:** compile error

**Actual behavior:** compile succeed

**Comment by MartinJohns** [comment]
[Why are functions with fewer parameters assignable to functions that take more parameters?](https://github.com/microsoft/TypeScript/wiki/FAQ#why-are-functions-with-fewer-parameters-assignable-to-functions-that-take-more-parameters)
